src/sentry/replays/consumers/recording_buffered.py

```diff
@@ -199,7 +199,7 @@ def process_message(buffer: RecordingBuffer, message: bytes) -> None:
             return None
 
     try:
-        headers, recording_data = process_headers(decoded_message["payload"])
+        headers, recording_data = process_headers(cast(bytes, decoded_message["payload"]))
```

Comment:
cast and type ignore should be reserved for bugs in the type checker

please fix the underlying annotations

Reply:
The issue is we're using a JSON schema validator for a msgpack payload.  msgpack accepts bytes.  JSON does not.  This worked previously because the underlying library implicitly typed it as `Any` but now that has been changed.

Reply:
@asottile-sentry the annotations are auto-generated from schema definitions, and due to limitations in that code generation we need to work around it here. we can fix the code generation but it is not trivial and they are outside of sentry

Reply:
we can probably add it as code comment but we have a bunch of places where we'd need one I think. maybe an explicit helper function for this particular kind of cast?

```
def cast_bytes(x: Any) -> bytes:
    """Long-winded explanation about sentry-kafka-schemas here"""
    return x
```